### PR TITLE
s3 initial setup, config and upload route

### PR DESCRIPTION
**Pull request by damienaries**
### What this PR does (required):
- Create "lovingdogsitter" aws S3 bucket and set up environment variables in .env
- Obtain bucket URL & create upload route function

### Screenshots / Videos (required):
Postman test
-Screen Shot 2021-08-16 at 9 44 48 PM
server/routes/upload.js
Screen Shot 2021-08-16 at 9 45 28 PM

### Any information needed to test this feature (required):
- See AWS keys in Slack channel
- run **npm install** to update dependencies list. I added multer, multer-s3, aws-sdk

### Any issues with the current functionality (optional):
- Still need to create routes and variables for different requests, and connect to front end

### CODE REVIEW

File: server/routes/upload.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            cb(null, { fieldName: file.fieldName });
+        },
+        key: function(req, file, cb) {
+            cb(null, Date.now().toString());
+        }
+    })
+})
+
+// upload route | Allow 5 images
+app.put(`/${s3URL}`, upload.array('photos', 5), (req, res, next) => {
```
Comment (rajivtitus) [inline server/routes/upload.js:25]: Maybe us an HTTP POST method here as we are creating a resource

File: server/routes/upload.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+const aws = require('aws-sdk');
+const express = require('express');
+const multer = require('multer');
+const multerS3 = require('multer-s3');
+import { config } from '../utils/s3Config';
+const s3URL = 'lovingdogsitter.s3.us-west-1.amazonaws.com/media';
```
Comment (rajivtitus) [inline server/routes/upload.js:6]: Once the route has been changed you can remove this variable too

**Review by rajivtitus — APPROVED** [review]
Looks good to start! Just a couple of minor comments

File: server/routes/upload.js (hunk 2 above)
Comment (damienaries) [reply server/routes/upload.js:6]: Done. So at which point do we set the url in the code? Or do we not need it since we have the aws config object?

**Comment by damienaries** [comment]
@rajivtitus I just pushed those changes, whenever you get a chance to check

File: server/routes/upload.js (hunk 2 above)
Comment (rajivtitus) [reply server/routes/upload.js]: I don't think we need this as we have the config object.

**Comment by rajivtitus** [comment]
> @rajivtitus I just pushed those changes, whenever you get a chance to check

LGTM! Thanks for making changes

**Comment by damienaries** [comment]
Thanks. Merging branch

**Pull request merged**